### ISSUE TITLE
Traffic splitting WIP

### ISSUE THREAD

**Issue by drnic** [cppforlife/knctl#27]
Re: https://github.com/cppforlife/knctl/blob/master/docs/traffic-splitting.md

What is the nature of "Currently does not work with latest Knative Serving"? Did this used to work but they changed the API?

**Comment by cppforlife** [comment]
related to https://github.com/knative/serving/issues/1812

currenlty knative service objects imply that knative controls all aspects of route objects that it creates. hence when user changes route obj to route to different revisions, knative will quickly revert it back. there is ongoing discussion about how to change knative to allow user control of route but also keeping service object around. (currently one can do this by delete service obj and keeping configuration + route objects around).

**Comment by cppforlife** [comment]
proposed cli ux:

```bash
knctl route manage --route hello # change route to use latest configuration
knctl route unmanage --route hello # change service to use `manual` key
knctl rollout --route hello -r hello:latest=5% -r hello:previous=95%
knctl rollout --route hello -r hello:latest=10% -r hello:previous=90%
knctl rollout --route hello -r hello:latest=50% -r hello:previous=50%
knctl rollout --route hello -r hello:latest=100% -r hello:previous=0%
knctl rollout --route hello -s hello=100% -s hello=0%
knctl rollout --route hello -s hello # equivalent of `knctl route manage --route hello`
knctl rollout --route hello --percent 5 # increase percentage of latest by X
```

**Comment by cppforlife** [comment]
implemented in v0.1.0 as knctl rollout command.
